### ISSUE TITLE
Feature request: SFTP/SSH environment variable(s)

### ISSUE THREAD

**Issue by ggets** [rclone/rclone#6094]
#### What is your current rclone version (output from `rclone version`)?
rclone v1.58.0
- os/version: Microsoft Windows 11 Pro 21H2 (64 bit)
- os/kernel: 10.0.22000.593 (x86_64)
- os/type: windows
- os/arch: amd64
- go/version: go1.17.8
- go/linking: dynamic
- go/tags: cmount

#### What problem are you are trying to solve?
Use `ssh -o 'SetEnv MYVAR=test' <host>` to set a environment variable on remote server.
#### How do you think rclone should be changed to solve that?
Either option to utilize `SendEnv` / `SetEnv` ... or better yet support for ssh options (like `ssh -o`) so other options can be used as well?

**Comment by ncw** [comment]
From the ssh docs

>      SendEnv
>              Specifies what variables from the local environ(7) should be sent to the server.  The server must also support it, and the server must be configured to accept these envi‐
>              ronment variables.  Note that the TERM environment variable is always sent whenever a pseudo-terminal is requested as it is required by the protocol.  Refer to AcceptEnv in
>              sshd_config(5) for how to configure the server.  Variables are specified by name, which may contain wildcard characters.  Multiple environment variables may be separated by
>              whitespace or spread across multiple SendEnv directives.
> 
>              See PATTERNS for more information on patterns.
> 
>              It is possible to clear previously set SendEnv variable names by prefixing patterns with -.  The default is not to send any environment variables.

>      SetEnv  Directly specify one or more environment variables and their contents to be sent to the server.  Similarly to SendEnv, the server must be prepared to accept the environment
>              variable.

Implementing an equivalent of `SendEnv` and `SetEnv` would probably be reasonably straight forward assuming the underlying library can do it:  https://pkg.go.dev/golang.org/x/crypto/ssh

It apparently does have support for the underlying primitive here: https://pkg.go.dev/golang.org/x/crypto/ssh#Session.Setenv

So this is mostly a matter of adding the configuration to rclone and calling that function. We should probably aim for `SetEnv` first and use `fs.SpaceSepList` to collect the environment variables in a `var=val var=val` format. This format can also be `"var=val with spaces"` also.

As for `SendEnv` we can use `filter` framework to build up something to match the wildcards quite easily. I don't think we need to keep exactly to the format used by ssh.

Is this something you'd like to work on @ggets ? If it is I can prepare instructions. If not then @albertony has done a lot of work on the sftp backend recently - maybe they would take a look?

**Comment by ggets** [comment]
Glad to hear it can be easily implemented. Sadly, I think it's outside the scope of my knowledge. GO just isn't my thing, I know nothing about it, sorry. Anything else I can help with, I will be glad to do so.

**Comment by ivandeex** [comment]
AFAIK rclone does not use SCP, but SFTP
SFTP subsystem of SSH daemons does not handle SendEnv
I doubt that proposed feature is of any use

**Comment by ggets** [comment]
Yes, It surely can be handled as both `SetEnv `and `SendEnv `are an official option:
https://man.openbsd.org/sftp

**Comment by ncw** [comment]
I've had a go at implementing `set_env`

[v1.59.0-beta.6062.c80b4d3af.fix-6094-sftp-env](https://beta.rclone.org/branch/fix-6094-sftp-env/v1.59.0-beta.6062.c80b4d3af.fix-6094-sftp-env/) on branch [fix-6094-sftp-env](https://github.com/rclone/rclone/tree/fix-6094-sftp-env) (uploaded in 15-30 mins)

I tested it was working like this

    rclone md5sum --max-depth 1 -vv ':sftp,user=sftptest,set_env="LC_POTATO=POTATO",md5sum_command=/tmp/env,sha1sum_command=/tmp/env:'normal-file.tx

Note that by default openssh is set up so that it doesn't accept many environment variables but `LC_*` is one thing it does accept.

#### --sftp-set-env

Environment variables to pass to sftp and commands

Set environment variables in the form:

    VAR=value

to be passed to the sftp client and to any commands run (eg md5sum).

Pass multiple variables space separated, eg

    VAR1=value VAR2=value

and pass variables with spaces in in quotes, eg

    "VAR3=value with space" "VAR4=value with space" VAR5=nospacehere



Properties:

- Config:      set_env
- Env Var:     RCLONE_SFTP_SET_ENV
- Type:        SpaceSepList
- Default:
